File: tests/test_parse/data/sample.m
function area = circle(r)
    area = pi * r^2;
end

function [area, perimeter] = rectangle(width, height)
    area = width * height;
    perimeter = 2 * (width + height);
end

% Class definition
classdef User
    properties
        name
        age
    end
    
    methods
        function obj = User(name, age)
            obj.name = name;
            obj.age = age;
        end
        
        function result = isAdult(obj)
            result = obj.age >= 18;
        end
    end
end 

File: tests/test_parse/data/sample.nim
type
  User = object
    name: string
    age: int

proc createUser(name: string, age: int): User =
  result = User(name: name, age: age)

proc isAdult(user: User): bool =
  result = user.age >= 18

proc formatUser(user: User): string =
  result = user.name & " (" & $user.age & " years old)" 